core/include/detray/core/mask_store.hpp

```diff
@@ -21,7 +21,7 @@ namespace detray {
 /** A mask store that provides the correct mask containers to client classes. */
 template <template <typename...> class tuple_t = dtuple,
           template <typename...> class vector_t = dvector,
-          typename... mask_types>
+          typename ID = unsigned int, typename... mask_types>
```

Comment:
Why do you need to parametrize `ID`? Can't it be just `unsigned int` always?

Reply:
The point is to enforce the use of the correct type ID with its corresponding container to make the code more explicit about what the linking does